client/state/document-head/reducer.js

```diff
@@ -47,17 +47,19 @@ export const meta = withSchemaValidation( metaSchema, ( state = DEFAULT_META_STA
 
 export const link = withSchemaValidation( linkSchema, ( state = [], action ) => {
 	switch ( action.type ) {
-		case DOCUMENT_HEAD_LINK_SET:
+		case DOCUMENT_HEAD_LINK_SET: {
 			if ( ! action.link ) {
 				return state;
 			}
 
 			// Append action.link to the state array and prevent duplicate objects.
 			// Works with action.link being a single link object or an array of link objects.
-			return uniqWith(
-				[ ...state, ...( Array.isArray( action.link ) ? action.link : [ action.link ] ) ],
-				isEqual
-			);
+			const links = Array.isArray( action.link ) ? action.link : [ action.link ];
+			state = links.reduce( ( accuState, newLink ) => {
+				const isNew = ! accuState.some( ( accuLink ) => isEqual( accuLink, newLink ) );
+				return isNew ? [ ...accuState, newLink ] : accuState;
```

Comment:
Just curious, is there a reason you create a new array every turn instead of e.g. pushing to `accuState`?

Reply:
I don't know how important it is, but the original code removed duplicate instances also when they both were in the `action.link` array. That's replicated by pushing to `state` one by one, and running the `state.some` loop over the full array with previously added items also considered.